recipe/sandbox/linear_algebra/lu_decomposition_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "recipe/sandbox/linear_algebra/lu_decomposition.h"
+#include <gtest/gtest.h>
+#include <cmath>
+#include <stdexcept>
+#include "recipe/sandbox/linear_algebra/matrix.h"
+#include "recipe/sandbox/linear_algebra/vector.h"
+
+namespace recipe {
+namespace sandbox {
+namespace linear_algebra {
+
+void EXPECT_EQ_VECTOR(const Vector& a, const Vector& b) {
+  EXPECT_EQ(a.Length(), b.Length());
+
+  for (int i = 0; i < a.Length(); i++) {
+    EXPECT_NEAR(a(i), b(i), 1e-6);
+  }
+}
+
+void EXPECT_NE_VECTOR_ELEM(const Vector& a, const Vector& b) {
+  EXPECT_EQ(a.Length(), b.Length());
+
+  bool ne_flg = false;
+  for (int i = 0; i < a.Length(); i++) {
+    if (abs(a(i) - b(i)) > 1e-6) ne_flg = true;
+  }
+
+  EXPECT_TRUE(ne_flg);
+}
+
+std::vector<std::tuple<Matrix, Vector, Vector>> TestCases() {
+  std::vector<std::tuple<Matrix, Vector, Vector>> vec;
+
+  // sample1
+  Matrix a1 = Matrix(3, 3);
+  a1(3, 3);
```

Comment:
Do you need this line? Besides, this line may causes segmentation fault. `assert` are useful to detect this kind of error.

```cpp
double Matrix::operator()(int i, int j) const {
assert(i < nrow && j < ncol);
return data[i * ncol + j]; 
}
```